core/src/main/java/com/graphhopper/routing/util/BikeCommonFlagEncoder.java

```diff
@@ -571,19 +589,34 @@ void collect( OSMWay way, double wayTypeSpeed, TreeMap<Double, Integer> weightTo
                     weightToPrioMap.put(40d, UNCHANGED.getValue());
             }
         } else if (avoidHighwayTags.contains(highway)
-                || maxSpeed >= avoidSpeedLimit && !"track".equals(highway))
-        {
-            weightToPrioMap.put(50d, REACH_DEST.getValue());
-            if (way.hasTag("tunnel", intendedValues))
-                weightToPrioMap.put(50d, AVOID_AT_ALL_COSTS.getValue());
-        }
+                    || maxSpeed >= avoidSpeedLimit && !"track".equals(highway))
+            {
+                weightToPrioMap.put(50d, REACH_DEST.getValue());
+                if (way.hasTag("tunnel", intendedValues))
+                    weightToPrioMap.put(50d, AVOID_AT_ALL_COSTS.getValue());
+            }
 
         if (pushingSectionsHighways.contains(highway)
                 || way.hasTag("bicycle", "use_sidepath")
                 || "parking_aisle".equals(service))
         {
-            if (way.hasTag("bicycle", "yes"))
-                weightToPrioMap.put(100d, UNCHANGED.getValue());
+            if (way.hasTag("bicycle", intendedValues))
```

Comment:
> From my perspective the code complexity remains the same compared to ther situation before the change. 

I mean the next 15 lines

> If the comment is meant as a general complaint about the complexity of the pushing section handling

It is really important for me that I do never "complain"! I was just asking. Of course complexity is sometimes necessary and here I wanted to find out if the difference for the route quality is really worth the effort of the nested ifs. E.g. there are two statements `weightToPrioMap.put(100d, VERY_NICE.getValue());` which could be merged to avoid nesting a bit.